doc/src/tutorials/biomechanics/biomechanics.rst

```diff
@@ -0,0 +1,799 @@
+
+::
+
+   >>> dadt = biceps.rhs().col_join(triceps.rhs())
+
+Evaluate the System Differential Equations
+==========================================
+
+The complete set of differential equations for this system take the form:
+
+.. math::
+
+   \begin{bmatrix}
+     \mathbf{I} & \mathbf{0} & \mathbf{0} \\
+     \mathbf{0} & \mathbf{M}_d &  \mathbf{0} \\
+     \mathbf{0} & \mathbf{0}   & \mathbf{I}
+   \end{bmatrix}
+   \begin{bmatrix}
+     \dot{\mathbf{q}} \\
+     \dot{\mathbf{u}} \\
+     \dot{\mathbf{a}}
+   \end{bmatrix}
+   =
+   \begin{bmatrix}
+     \mathbf{u} \\
+     \mathbf{g}_d(\mathbf{q}, \mathbf{u}, \mathbf{a})  \\
+     \mathbf{g}_a(\mathbf{a}, \mathbf{e})
+   \end{bmatrix}
+
+In this case, only the dynamical differential equations require solving the
+linear system to put into explicit form.
+
+To evaluate the system's equations we first need to gather up all of the state,
+input, and constant variables for use with
+:obj:`~sympy.utilities.lambdify.lambdify`. The state vector is made up of the
+coordinates, generalized speeds, and the two muscles' activation state:
+:math:`\mathbf{x}=\begin{bmatrix}\mathbf{q}\\\mathbf{u}\\\mathbf{a}\end{bmatrix}`.
+
+::
+
+   >>> q, u = kane.q, kane.u
+   >>> a = biceps.x.col_join(triceps.x)
+   >>> x = q.col_join(u).col_join(a)
+   >>> x
+   Matrix([
+   [       q1(t)],
+   [       q2(t)],
+   [       q3(t)],
+   [       q4(t)],
+   [       u1(t)],
+   [       u2(t)],
+   [       u3(t)],
+   [       u4(t)],
+   [ a_biceps(t)],
+   [a_triceps(t)]])
+
+The only specific inputs are the two muscles' excitation:
+
+::
+
+   >>> e = biceps.r.col_join(triceps.r)
+   >>> e
+   Matrix([
+   [ e_biceps(t)],
+   [e_triceps(t)]])
+
+The constants are made up of the geometry, mass, local gravitational constant,
+the lever's stiffness and damping coefficients, and various parameters of the
+muscles.
+
+::
+
+   >>> p = sm.Matrix([
+   ...     dx,
+   ...     dy,
+   ...     dz,
+   ...     lA,
+   ...     lC,
+   ...     lD,
+   ...     mA,
+   ...     mC,
+   ...     mD,
+   ...     g,
+   ...     k,
+   ...     c,
+   ...     r,
+   ...     biceps._F_M_max,
+   ...     biceps._l_M_opt,
+   ...     biceps._l_T_slack,
+   ...     biceps._v_M_max,
+   ...     biceps._alpha_opt,
+   ...     biceps._beta,
+   ...     triceps._F_M_max,
+   ...     triceps._l_M_opt,
+   ...     triceps._l_T_slack,
+   ...     triceps._v_M_max,
+   ...     triceps._alpha_opt,
+   ...     triceps._beta,
```

Comment:
Seems like these should be public and settable. Either you should expect someone to always create symbols for these and pass them in (so they have a copy of the symbols when needed) or they need to be easily accessible if the user doesn't pass them in. Probably both in this case.

Reply:
These are public, so remover underscores.